Library/Homebrew/cmd/bottle.rb

```diff
@@ -183,6 +186,15 @@ def bottle_formula(f)
     relocatable = false
     skip_relocation = false
 
+    formula_source_time = f.stable.stage do
+      newest_mtime = nil
+      Pathname(Dir.pwd).find do |path|
+        newest_mtime ||= path.mtime
+        newest_mtime = path.mtime if path.mtime > newest_mtime
+      end
+      newest_mtime
+    end
```

Comment:
For performance(to avoid unnecessary stage) and consistency(whether bottle or not), it would be better to put these logic in `formula_installer`. And here we could just read time from any file in keg(except tab file).